testsuite/smoke-test/src/aptos_cli.rs

```diff
@@ -211,6 +169,185 @@ async fn test_register_and_update_validator() {
+async fn test_join_and_leave_validator() {
+    let (mut swarm, cli, _faucet) = SwarmBuilder::new_local(1)
+        .with_aptos()
+        .with_init_config(Arc::new(|_i, conf, genesis_stake_amount| {
+            // reduce timeout, as we will have dead node during rounds
+            conf.consensus.round_initial_timeout_ms = 100;
+            conf.consensus.quorum_store_poll_count = 2;
+            *genesis_stake_amount = 100000;
+        }))
+        .with_init_genesis_config(Arc::new(|genesis_config| {
+            genesis_config.allow_new_validators = true;
+            genesis_config.epoch_duration_secs = 3600;
+            genesis_config.min_price_per_gas_unit = 0;
+        }))
+        .build_with_cli(0)
+        .await;
+
+    let transaction_factory = swarm.chain_info().transaction_factory();
+    let rest_client = swarm.validators().next().unwrap().rest_client();
+
+    let mut keygen = KeyGen::from_os_rng();
+    let validator_cli_index = 0;
+    let keys = init_validator_account(&cli, &mut keygen, validator_cli_index).await;
+    let mut gas_used = 0;
+
+    // faucet can make our root LocalAccount sequence number get out of sync.
+    swarm
+        .chain_info()
+        .resync_root_account_seq_num(&rest_client)
+        .await
+        .unwrap();
+
+    let port = 1234;
+    gas_used += get_gas(
+        cli.register_validator_candidate(
+            validator_cli_index,
+            keys.consensus_public_key(),
+            keys.consensus_proof_of_possession(),
+            HostAndPort {
+                host: dns_name("0.0.0.0"),
+                port,
+            },
+            keys.network_public_key(),
+        )
+        .await
+        .unwrap(),
+    );
+
+    assert_validator_set_sizes(&cli, validator_cli_index, 1, 0, 0).await;
+
+    assert_eq!(
+        DEFAULT_FUNDED_COINS - gas_used,
+        cli.account_balance(validator_cli_index).await.unwrap()
+    );
+
+    let stake_coins = 7;
+    gas_used += get_gas(
+        cli.add_stake(validator_cli_index, stake_coins)
+            .await
+            .unwrap(),
+    );
+
+    assert_eq!(
+        DEFAULT_FUNDED_COINS - stake_coins - gas_used,
+        cli.account_balance(validator_cli_index).await.unwrap()
+    );
+
+    reconfig(
+        &rest_client,
+        &transaction_factory,
+        swarm.chain_info().root_account(),
+    )
+    .await;
+
+    assert_validator_set_sizes(&cli, validator_cli_index, 1, 0, 0).await;
+
+    let lockup_duration = Duration::from_secs(20);
+    gas_used += get_gas(
+        cli.increase_lockup(validator_cli_index, lockup_duration)
+            .await
+            .unwrap(),
+    );
+
+    let lockup_start = SystemTime::now();
+
+    reconfig(
+        &rest_client,
+        &transaction_factory,
+        swarm.chain_info().root_account(),
+    )
+    .await;
+
+    assert_validator_set_sizes(&cli, validator_cli_index, 1, 0, 0).await;
+
+    gas_used += get_gas(cli.join_validator_set(validator_cli_index).await.unwrap());
```

Comment:
Do we want to test that the validator can now vote after joining the validator set?

Reply:
I am not actually creating a validator node. I am having some issues with private keys matching. 

I think this test covers enough as it is, and we can have a separate test for that.